include/glow/Graph/Graph.h

```diff
@@ -163,9 +163,27 @@ class Module final {
   /// Get the pseudo-random number generator used by this module.
   PseudoRNG &getPRNG() { return PRNG_; }
 
-  /// Dumps the textual representation of the network.
+  /// Dump a textual representation of the Module into default output stream.
   void dump() const;
 
+  /// Dump a textual representation of the Module to std::string.
+  std::string toString() const;
+
+  /// Dump a textual representation of the Module into provided output stream.
+  void dump(llvm::raw_ostream &os) const;
+
+  friend llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
```

Comment:
Is there a need to declare the `operator <<` inside the class as a friend? It seems like in Glow we declare it usually define such operators outside of the class. See e.g. `Type.h`.

More over, it probably makes sense to move the implementation into the cpp file.

Reply:
> Is there a need to declare the `operator <<` inside the class as a friend? It seems like in Glow we declare it usually define such operators outside of the class. See e.g. `Type.h`.
> 
> More over, it probably makes sense to move the implementation into the cpp file.

OK. I switched them from friend function to outside and moved the implementations in the cpp file.